### DISCUSSION TITLE
Gradle jvmargs property from gradle.properties is not recognized in codemagic build

### DISCUSSION THREAD

**Discussion by nateshmbhat** [codemagic-ci-cd/codemagic-docs#2427] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

React Native

### Steps to reproduce

Though i have specified the jvmargs in my gradle properties file , `org.gradle.jvmargs=-Xmx4g -XX:MaxMetaspaceSize=2g` , i'm getting error during my android build that 

```
The Daemon will expire after the build after running out of JVM heap space.
The project memory settings are likely not configured or are configured to an insufficient value.
The daemon will restart for the next build, which may increase subsequent build times.
These settings can be adjusted by setting 'org.gradle.jvmargs' in 'gradle.properties'.
The currently configured max heap space is '2 GiB' and the configured max metaspace is 'unknown'.
For more information on how to set these values, please refer to https://docs.gradle.org/8.3/userguide/build_environment.html#sec:configuring_jvm_memory in the Gradle documentation.
To disable this warning, set 'org.gradle.daemon.performance.disable-logging=true'.
Daemon will be stopped at the end of the build after running out of JVM heap space

> Task :app:minifyProdReleaseWithR8 FAILED
ERROR: R8: java.lang.OutOfMemoryError: Java heap space
```

> The currently configured max heap space is '2 GiB' and the configured max metaspace is 'unknown'.

codemagic is not able to recognize the maxheap space and max meta space config that i have given in gradle.properties file. 
So i tried running gradle with debug option and found below logs which say that env project properties and system project properties and empty 🤔 

```
2024-03-22T08:23:12.823+0000 [DEBUG] [org.gradle.internal.operations.DefaultBuildOperationRunner] Build operation 'Load build' started
2024-03-22T08:23:12.831+0000 [DEBUG] [org.gradle.initialization.properties.DefaultProjectPropertiesLoader] Found env project properties: []
2024-03-22T08:23:12.831+0000 [DEBUG] [org.gradle.initialization.properties.DefaultProjectPropertiesLoader] Found system project properties: []
```

how do i resolve this issue ? 

### Expected results

Gradle should not say : 'The currently configured max heap space is '2 GiB' and the configured max metaspace is 'unknown''
This means it will use the value of jvmargs from the gradle.properties file in our project

### Actual results

Gradle shows this warning before throwing error that it ran out of jvm heap space.
The currently configured max heap space is '2 GiB' and the configured max metaspace is 'unknown'

### Build id (optional)

65fc5c0e5e7972cb09c330fd

**Comment by dtrdic** [comment]
hey @nateshmbhat, to resolve this Java heap error, let's try the solution that has helped different users as well. Let's allocate some memory for the heap space by modifying your `android/gradlew` file and adding this command:

`exec "$JAVACMD" "${JVM_OPTS[@]}" -classpath "$CLASSPATH" org.gradle.wrapper.GradleWrapperMain -Dorg.gradle.jvmargs="-Xmx4096m -XX:MaxPermSize=1024m" "$@"`

In this example, I have allocated 4096M(4GB) for the heap space, this should ensure that enough memory is allocated for the build process
<img width="1071" alt="image" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/38918425/0d20f33f-3435-4021-9666-a44652c72911">

**Comment by dtrdic** [comment]
I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new discussion.

**Comment by ahmedserag2** [comment]
@dtrdic i know this has been stale for quite sometime but i had a similar issue with my android flutter build. after abit of tinkering with the exec command you provided it seems to have done the trick.. however i still dont get why the specs i provided in my gradle.properties file didnt reflect needing to be provided from gradlew bash script. is there a reason the codemagic runner doesnt recognize the gradle.properties file ?
